db/20200528140124_add_cvat_workspace_template.go

```diff
@@ -0,0 +1,164 @@
+  env:
+  - name: DJANGO_MODWSGI_EXTRA_ARGS
+    value: ""
+  - name: ALLOWED_HOSTS
+    value: '*'
+  - name: CVAT_REDIS_HOST
+    value: localhost
+  - name: CVAT_POSTGRES_HOST
+    value: localhost
+  - name: CVAT_SHARE_URL
+    value: /home/django/data
+  ports:
+  - containerPort: 8080
+    name: http
+  volumeMounts:
+  - name: data
+    mountPath: /home/django/data
+  - name: keys
+    mountPath: /home/django/keys
+  - name: logs
+    mountPath: /home/django/logs
+  - name: models
+    mountPath: /home/django/models
+- name: cvat-ui
+  image: onepanel/cvat-ui:v0.6.24
+  ports:
+  - containerPort: 80
+    name: http
+ports:
+- name: cvat-ui
+  port: 80
+  protocol: TCP
+  targetPort: 80
+- name: cvat
+  port: 8080
+  protocol: TCP
+  targetPort: 8080
+routes:
+- match:
+  - uri:
+      regex: /api/.*|/git/.*|/tensorflow/.*|/auto_annotation/.*|/analytics/.*|/static/.*|/admin/.*|/documentation/.*|/dextr/.*|/reid/.*
+  - queryParams:
+      id:
+        regex: \d+.*
+  route:
+  - destination:
+      port:
+        number: 8080
+- match:
+  - uri:
+      prefix: /
+  route:
+  - destination:
+      port:
+        number: 80
+postExecutionWorkflow:
+  entrypoint: main
+  templates:
+  - name: main
+    dag:
+       tasks:
+       - name: slack-notify
+         template: slack-notify
+  -  name: slack-notify
+     container:
+       image: technosophos/slack-notify
+       args:
+       - SLACK_USERNAME=onepanel SLACK_TITLE="Your workspace is ready" SLACK_ICON=https://www.gravatar.com/avatar/5c4478592fe00878f62f0027be59c1bd SLACK_MESSAGE="Your workspace is now running" ./slack-notify
+       command:
+       - sh
+       - -c
+`
+
+const cvatTemplateName = "CVAT"
+
+func init() {
+	goose.AddMigration(Up20200528140124, Down20200528140124)
+}
+
+func Up20200528140124(tx *sql.Tx) error {
+	// This code is executed when the migration is applied.
+
+	client, err := getClient()
+	if err != nil {
+		return err
+	}
+
+	namespaces, err := client.ListOnepanelEnabledNamespaces()
+	if err != nil {
+		return err
+	}
+
+	workspaceTemplate := &v1.WorkspaceTemplate{
+		Name:     cvatTemplateName,
+		Manifest: cvatWorkspaceTemplate,
+	}
+
+	for _, namespace := range namespaces {
+		if _, err := client.CreateWorkspaceTemplate(namespace.Name, workspaceTemplate); err != nil {
+			log.Printf("error %v", err.Error())
+		}
+	}
+
+	return nil
+}
+
+func Down20200528140124(tx *sql.Tx) error {
+	// This code is executed when the migration is rolled back.
+
+	client, err := getClient()
+	if err != nil {
+		return err
+	}
+
+	namespaces, err := client.ListOnepanelEnabledNamespaces()
+	if err != nil {
+		return err
+	}
+
+	for _, namespace := range namespaces {
+		if err := client.DeleteWorkspace(namespace.Name, cvatTemplateName); err != nil {
```

Comment:
@aleksandrmelnikov @Vafilor is this supposed to be `client.ArchiveWorkspaceTemplate`? If so, the JupyterLab one needs to be fixed too.

Also note the changes I made in the this PR: https://github.com/onepanelio/core/pull/260/commits/7b570f861ef504ae8f5ef8e0593ae9e3901a22a1

Reply:
Two thoughts:
- If we want to support running the "down" migration after they make DB / K8S changes (or they've done some work), then yes. I think we need to make this change.
- If they do this immediately on creation, then this is fine.

Reply:
Pushed up a possible change.